pytorch_pretrained_bert/tests/model_tests_commons.py

```diff
@@ -77,6 +77,46 @@ def _create_and_check_torchscript(tester, model_classes, config, inputs_dict):
 
         tester.parent.assertTrue(models_equal)
 
+def _create_and_check_torchscript_trace(tester, model_classes, config, inputs_dict):
+    inputs = inputs_dict["input_ids"]
+    inputs_reduced = inputs[:int(inputs.shape[0]/2), ..., :int(inputs.shape[-1]/2)]
+    configs_no_init = _config_zero_init(config)  # To be sure we have no Nan
+    configs_no_init.torchscript = True
+    for model_class in model_classes:
+        model = model_class(config=configs_no_init)
+        model.eval()
+
+        try:
+            traced_model = torch.jit.trace(model, inputs)
+            traced_model_reduced = torch.jit.trace(model, inputs_reduced)
+
+            torch.jit.save(traced_model, "traced_model.pt")
+            torch.jit.save(traced_model_reduced, "traced_model_reduced.pt")
```

Comment:
It's not exactly what we should do. The weights won't change with a reduced input but the outputs will not be the same (or even break) if one model is stuck with a predefined batch.seq_length.
You should make a forward of the large inputs (so it can break the reduced model) in the normal and reduced model and compare the outputs.